src/gui/tray/unifiedsearchresultslistmodel.cpp

```diff
@@ -324,26 +324,46 @@ bool UnifiedSearchResultsListModel::isSearchInProgress() const
     return !_searchJobConnections.isEmpty();
 }
 
-void UnifiedSearchResultsListModel::resultClicked(const QString &providerId, const QUrl &resourceUrl) const
+
+void UnifiedSearchResultsListModel::resultClicked(
+    const QString &providerId, const QUrl &resourceUrl, const QString &subline, const QString &title
+) const
 {
     const QUrlQuery urlQuery{resourceUrl};
-    const auto dir = urlQuery.queryItemValue(QStringLiteral("dir"), QUrl::ComponentFormattingOption::FullyDecoded);
-    const auto fileName =
-        urlQuery.queryItemValue(QStringLiteral("scrollto"), QUrl::ComponentFormattingOption::FullyDecoded);
+    QString dir = urlQuery.queryItemValue(QStringLiteral("dir"), QUrl::ComponentFormattingOption::FullyDecoded);
+    QString fileName = urlQuery.queryItemValue(QStringLiteral("scrollto"), QUrl::ComponentFormattingOption::FullyDecoded);
 
-    if (providerId.contains(QStringLiteral("file"), Qt::CaseInsensitive) && !dir.isEmpty() && !fileName.isEmpty()) {
-        if (!_accountState || !_accountState->account()) {
+    if (providerId.contains(QStringLiteral("file"), Qt::CaseInsensitive)){
+        if (_accountState == nullptr || _accountState->account() == nullptr) {
```

Comment:
please revert
the reader should not need to know that what you test are naked pointers
they could or maybe are smart pointers
it is good practice in c++ to test the validity of a pointer using the version of the code that you have modified (`if (!ptr) {`)
```suggestion
        if (!_accountState || !_accountState->account()) {
```